Sources/Sentry/SentryFileManager.m

```diff
@@ -147,6 +150,41 @@ - (NSString *)storeEnvelope:(SentryEnvelope *)envelope {
     }
 }
 
+- (void)storeCurrentSession:(SentrySession *)session {
+    NSData *sessionData = [SentrySerialization dataWithSession:session options:0 error:nil];
+    [SentryLog logWithMessage:[NSString stringWithFormat:@"Writing to current session: %@", self.currentSessionFilePath] andLevel:kSentryLogLevelDebug];
+    @synchronized (self.currentSessionFilePath) {
+        [sessionData writeToFile:self.currentSessionFilePath options:NSDataWritingAtomic error:nil];
+    }
+}
+
+- (void)deleteCurrentSession {
+    [SentryLog logWithMessage:[NSString stringWithFormat:@"Deleting current session: %@", self.currentSessionFilePath] andLevel:kSentryLogLevelDebug];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    @synchronized (self.currentSessionFilePath) {
+        [fileManager removeItemAtPath:self.currentSessionFilePath error:nil];
+    }
+}
+
+- (SentrySession *_Nullable)readCurrentSession {
+    [SentryLog logWithMessage:[NSString stringWithFormat:@"Reading from current session: %@", self.currentSessionFilePath] andLevel:kSentryLogLevelDebug];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSData *currentSessionData = nil;
+    @synchronized (self.currentSessionFilePath) {
+        currentSessionData = [fileManager contentsAtPath:self.currentSessionFilePath];
+        if (nil == currentSessionData) {
+            return nil;
+        }
+        SentrySession *currentSession = [SentrySerialization sessionWithData:currentSessionData];
+        if (nil == currentSession) {
+            [SentryLog logWithMessage:[NSString stringWithFormat:@"Data stored in current session: '%@' was not parsed as session.",
+                    self.currentSessionFilePath] andLevel:kSentryLogLevelError];
+            return nil;
+        }
+        return currentSession;
+    }
+}
+
```

Comment:
What about putting all this functionality in an extra class? All those three methods belong together and don't have anything to do with the rest of this class here.

Reply:
Agree